### PR TITLE
Remodelling the documentation

### LINKED ISSUE #460: Restructure the documentation
Miro board: https://miro.com/app/board/uXjVK00vmT0=/

### LINKED ISSUE #463: Add missing/edit ambiguous docstrings
Missing docstrings:
- classes
    - `AnnotationsLoader`
- functions
    - `print_version`

Ambiguous docstrings:
- `PatchDataset` / `PatchContextDataset` = have the same summary string

### PR DESCRIPTION
**Pull request by kallewesterling**
### Summary

The documentation had unclear ordering and files that were named in ways that didn't optimise the SEO of our documentation. That should be fixed with this (see left image below, compared to the former structure on the right).

image

Fixes #460
Fixes #463

### Describe your changes

- minor edits of language for consistency
- fixing front page (removing the TODO list)
- remodelling the documentation files (moving quite a bit around!)

### Checklist before assigning a reviewer (update as needed)

- [ ] Self-review code
- [ ] Ensure submission passes current tests
- [ ] Add tests
- [ ] Update relevant docs

### Reviewer checklist

Please add anything you want reviewers to specifically focus/comment on.

- [ ] Everything looks ok?

### CODE REVIEW

**Review by rwood-97 — APPROVED** [review]
I built this here to check: https://mapreader.readthedocs.io/en/kallewesterling-issue460/

I realise now my comment re. numbering might not be a problem since the numbers don't show in the docs. Might be useful still to indicate the workflows for classification vs text.

Thank you!

File: docs/source/using-mapreader/step-by-step-guide/index.rst
[the remark this answers is not in the document]
Comment (kallewesterling) [reply docs/source/using-mapreader/step-by-step-guide/index.rst:1]: right, so you mean that:
- the download step is sort-of optional (as long as you have downloaded the maps already)
- the post-processing step is an optional tag-on?

[the remark this answers is not in the document]
Comment (rwood-97) [reply docs/source/using-mapreader/step-by-step-guide/index.rst:1]: yep and if you are doing text spotting you don't *have* to do classify too

**Comment by rwood-97** [comment]
I had a go at fixing merge conflicts here, mostly around the `coding basics` docs. Can you double check but then I think we are good to merge?

**Comment by kallewesterling** [comment]
Sounds good -- let me look at [your comment](https://github.com/Living-with-machines/MapReader/pull/462#discussion_r1686366591) before we do though!

**Comment by kmcdono2** [comment]
This looks great to me! Prompts me to ask @kallewesterling if we can set up some kind of reminder system to add to the 'project cv' and other elements that require updates.

**Comment by kallewesterling** [comment]
I think we should just have that as a standing point in our sprint meeting to keep revisiting. I'll add it to my idea board for Thursday meetings

**Pull request merged**